src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -161,33 +191,52 @@ def __call__(
                 device = "cuda" if torch.cuda.is_available() else "cpu"
             self.to(device)
 
+        # enable/disable grad
+        was_grad_enabled = torch.is_grad_enabled()
```

Comment:
This will also be a bit difficult to accept for me. It's a) a bit hacky to me and b) Pipelines by definition should only be used for inference. I assume the gradients are needed for analysis and the idea is not to do training? 
I'm not 100% sure whether they are enough use cases that warrant allowing gradient flow here on the other hand it also shouldn't hurt really if we leave the default to `False`. IMO working with function decorators and `enable_grad` + `disable_grad` functions  is the way to go here though instead.

What do you think @patil-suraj @anton-l @pcuenca ?

Reply:
Same as Patrick, not really in favor of this.

> IMO working with function decorators and enable_grad + disable_grad functions is the way to go here though instead.

+1

Reply:
> IMO working with function decorators and enable_grad + disable_grad functions is the way to go here though instead.

By that means, keeping `@torch.no_grad` decorator in `__call__` and add put the whole method under `with torch.enable_grad() if enable_grad else nullcontext():` ?
Not sure what you meant.